daemon/logger/splunk/splunk.go

```diff
@@ -301,7 +490,15 @@ func (l *splunkLogger) postMessage(message *splunkMessage) error {
 }
 
 func (l *splunkLogger) Close() error {
-	l.transport.CloseIdleConnections()
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	if l.closedCond == nil {
+		l.closedCond = sync.NewCond(&l.lock)
+		close(l.stream)
+		for !l.closed {
```

Comment:
I haven't used condition variables in Go -- from the use of the additional bool here I'm guessing we're protecting against spurious wake-ups?